Pharmacy Merger

mazinskihenry opened the pull request:
Full KAT Pharmacy Merger 
- All KAT Medications brought over (except ondansetron, has been removed for redundency purposes)
- Cardiac rhythm system brought over
- IV system brought over (ACE Pharmacy, not KAT Pharmacy version)
- Vehicle AED-X usage brought over

SPO2 Update
- Point based system, will move more dynamically
- Groundwork for O2 canisters

mazinskihenry committed:
Starter Merge

mazinskihenry committed:
Pharmacy Merger + SPO2 Update

Tomcat-SG commented on addons/airway/functions/fnc_updateInjuryList.sqf:
```diff
@@ -173,6 +162,56 @@ if (_target getVariable ["KAT_medical_tensionpneumothorax", false] && _selection
+private _placed = _target getVariable [QEGVAR(pharma,IVplaced), false];
+private _site = _target getVariable [QEGVAR(pharma,IVsite), 0];
+
+if ((_placed == true) && (_site == _selectionN)) then {
```
```suggestion
if (_placed && {_site == _selectionN}) then {
```

Tomcat-SG commented on addons/airway/functions/fnc_updateInjuryList.sqf:
```diff
@@ -173,6 +162,56 @@ if (_target getVariable ["KAT_medical_tensionpneumothorax", false] && _selection
+if ((_placed == true) && (_site == _selectionN)) then {
+    if (_site > 1) then {
+        private _a = "IV_16";
+        private _text = format ["STR_kat_pharma_%1_Display", _a];
```
```suggestion
```

Tomcat-SG commented on addons/airway/functions/fnc_updateInjuryList.sqf:
```diff
@@ -173,6 +162,56 @@ if (_target getVariable ["KAT_medical_tensionpneumothorax", false] && _selection
+
+if ((_placed == true) && (_site == _selectionN)) then {
+    if (_site > 1) then {
+        private _a = "IV_16";
```
```suggestion
        private _text = format ["STR_kat_pharma_%1_Display", "IV_16"];
```

Tomcat-SG commented on addons/airway/functions/fnc_updateInjuryList.sqf:
```diff
@@ -173,6 +162,56 @@ if (_target getVariable ["KAT_medical_tensionpneumothorax", false] && _selection
+        _woundEntries pushback [localize _text, [0.3, 0.3, 0.5, 1]];
+    } else {
+        private _a = "IO_45";
+        private _text = format ["STR_kat_pharma_%1_Display", _a];
```
```suggestion
```

Tomcat-SG commented on addons/airway/functions/fnc_updateInjuryList.sqf:
```diff
@@ -173,6 +162,56 @@ if (_target getVariable ["KAT_medical_tensionpneumothorax", false] && _selection
+        private _text = format ["STR_kat_pharma_%1_Display", _a];
+        _woundEntries pushback [localize _text, [0.3, 0.3, 0.5, 1]];
+    } else {
+        private _a = "IO_45";
```
```suggestion
        private _text = format ["STR_kat_pharma_%1_Display", "IO_45"];
```

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        icon = QPATHTOF(ui\X_Series-Device_W.paa);
+    };
+    class Defibrillator_AED_X_vehicle: Defibrillator {
+        displayName = "Vehicle: Use AED-X";
```
localize for multi language

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        icon = QPATHTOF(ui\X_Series-Device_W.paa);
+    };
+    class X_Defibrillator_AED_X_vehicle: Defibrillator_AED_X_vehicle {
+        displayName = "Vehicle: Attach AED-X";
```
localize for multi language

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        icon = QPATHTOF(ui\X_Series-Device_W.paa);
+    };
+    class Remove_X_Defibrillator_vehicle: Defibrillator_AED_X_vehicle {
+        displayName = "Vehicle: Remove AED-X";
```
localize for multi language

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,47 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Handle the X Series Defi for the patient.
```
Is this right?

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,47 @@
+ * None
+ *
+ * Example:
+ * [player, cursorTarget] call kat_circulation_fnc_treatmentAdvanced_X;
```
this isnt right

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,47 @@
+ * 1: Target <OBJECT>
+ *
+ * Return Value:
+ * None
```
this isnt right

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,47 @@
+private _bloodLoss = _patient getVariable ["ace_medical_bloodVolume", 6.0];
+private _asystole = _patient getVariable [QGVAR(asystole), 0];
+
+if (_asystole == 0) then {
```
```suggestion
if (_asystole isEqualTo 0) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,47 @@
+playsound3D [QPATHTOF_SOUND(sounds\analyse.wav), _patient, false, getPosASL _patient, 5, 1, 15];
+
+// wait for the analyse and give the advise
+if (_patient getVariable ["ace_medical_heartRate", 0] == 0 && _asystole <= 1) then {
```
```suggestion
if (_patient getVariable ["ace_medical_heartRate", 0] isEqualTo 0 && {_asystole <= 1}) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -24,9 +24,51 @@ params [
+private _asystole = _patient getVariable [QGVAR(asystole), 0];
+private _CPRcount = _patient getVariable [QGVAR(CPRcount), 0];
+
+if (_asystole == 0) then {
```
```suggestion
if (_asystole isEqualTo 0) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -24,9 +24,51 @@ params [
+{
+    _x params ["_medication"];
+
+    switch(_medication) do
```
beautify the line spacing for the following switch

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -61,16 +103,33 @@ switch (_reviveObject) do {
 if (_reviveObject isEqualTo "AED" || _reviveObject isEqualTo "AED-X" || _reviveObject isEqualTo "AED-Station") exitWith {
-	if (_random <= _chance) then {
+	_chance = _chance + (_amiBoost + _lidoBoost * _epiBoost);
+	if (_random <= _chance && _asystole <= 1) then {
```
```suggestion
	if (_random <= _chance && {_asystole <= 1}) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -61,16 +103,33 @@ switch (_reviveObject) do {
 		["ace_medical_CPRSucceeded", _patient] call CBA_fnc_localEvent;
 	};
 } else {
+	diag_log "MANUAL CPR";
```
For release this should be removed

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -61,16 +103,33 @@ switch (_reviveObject) do {
 } else {
+	diag_log "MANUAL CPR";
+
+	if (_epiBoost == 1) then {
```
```suggestion
	if (_epiBoost isEqualTo 1) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_treatment.sqf:
```diff
@@ -166,8 +166,6 @@ if (_callbackProgress isEqualTo {}) then {
 
 [_medic, _patient, _bodyPart, _classname, _itemUser, _usedItem] call _callbackStart;
 
-["ace_treatmentStarted", [_medic, _patient, _bodyPart, _classname, _itemUser, _usedItem]] call CBA_fnc_localEvent;
```
why?

Tomcat-SG commented on addons/pharma/licence.txt:
```diff
@@ -0,0 +1,10 @@
+﻿This addon is made by Katalam.
```
Pharma Author name is Katalam? I dont think so...

Tomcat-SG commented on addons/circulation/functions/fnc_treatmentAdvanced_X.sqf:
```diff
@@ -32,11 +32,22 @@ _target setVariable [QGVAR(X), true, true];
+private _bloodLoss = _target getVariable ["ace_medical_bloodVolume", 6.0];
+private _asystole = _target getVariable [QGVAR(asystole), 0];
+
+if (_asystole == 0) then {
```
```suggestion
if (_asystole isEqualTo 0) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_treatmentAdvanced_X.sqf:
```diff
@@ -32,11 +32,22 @@ _target setVariable [QGVAR(X), true, true];
 
 // wait for the analyse and give the advise
-if (_target getVariable ["ace_medical_heartRate", 0] == 0) then {
+if ((_target getVariable ["ace_medical_heartRate", 0] == 0) && (_target getVariable [QGVAR(asystole), 0] < 2)) then {
```
```suggestion
if ((_target getVariable ["ace_medical_heartRate", 0] isEqualTo 0) && {_target getVariable [QGVAR(asystole), 0] < 2}) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_treatmentAdvanced_X_Vehicle.sqf:
```diff
@@ -0,0 +1,112 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Handle the X Series Defi for the patient.
```
```suggestion
 * Handle the X Series Defi for the patient in a Vehicle
```

Tomcat-SG commented on addons/circulation/functions/fnc_treatmentAdvanced_X_Vehicle.sqf:
```diff
@@ -0,0 +1,112 @@
+ * None
+ *
+ * Example:
+ * [player, cursorTarget] call kat_circulation_fnc_treatmentAdvanced_X;
```
not correct

Tomcat-SG commented on addons/circulation/functions/fnc_treatmentAdvanced_X_Vehicle.sqf:
```diff
@@ -0,0 +1,112 @@
+playsound3D [QPATHTOF_SOUND(sounds\analyse.wav), _target, false, getPosASL _target, 5, 1, 15];
+
+// wait for the analyse and give the advise
+if (_target getVariable ["ace_medical_heartRate", 0] == 0) then {
```
```suggestion
if (_target getVariable ["ace_medical_heartRate", 0] isEqualTo 0) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_vehicleCheck.sqf:
```diff
@@ -0,0 +1,46 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "LeftLeg", "", objNull, "kat_IV_20"] call kat_circulation_fnc_applyIV;
```
not correct

Tomcat-SG commented on addons/circulation/functions/fnc_vehicleCheck.sqf:
```diff
@@ -0,0 +1,46 @@
+
+    if (_x isKindOf "LandVehicle" || _x isKindOf "Helicopter") then {
+        {
+            if (_x == "kat_X_AED") then {
```
```suggestion
            if (_x isEqualTo "kat_X_AED") then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_vehicleCheck.sqf:
```diff
@@ -0,0 +1,46 @@
+            };
+        } forEach _inventory;
+
+        if (_return == true) then {
```
```suggestion
        if (_return) then {
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        items[] = {"kat_norepinephrine"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
+        patientStateCondition = 0;
+        callbackSuccess = "ace_medical_treatment_fnc_medication";
```
Missing params for this function of ACE3

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        items[] = {"kat_phenylephrine"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
+        patientStateCondition = 0;
+        callbackSuccess = "ace_medical_treatment_fnc_medication";
```
Missing params for this function of ACE3

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        items[] = {"kat_nitroglycerin"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
+        patientStateCondition = 0;
+        callbackSuccess = "ace_medical_treatment_fnc_medication";
```
Missing params for this function of ACE3

Tomcat-SG commented on addons/pharma/CfgFunctions.hpp:
```diff
@@ -0,0 +1,32 @@
+    class overwrite_ace_medical_gui {
+        tag = "ace_medical_gui";
+        class ace_medical_gui {
+            class updateInjuryList {
```
why this is in the pharma addon? this is made already in the airway addon folder.

Tomcat-SG commented on addons/pharma/CfgFunctions.hpp:
```diff
@@ -0,0 +1,32 @@
+	class overwrite_medical_treatment {
+		tag = "ace_medical_treatment";
+		class ace_medical_treatment {
+            class cprLocal {
```
why this is in the pharma addon? this is made already in the circulation addon folder.

Tomcat-SG commented on addons/pharma/CfgFunctions.hpp:
```diff
@@ -0,0 +1,32 @@
+                file = QPATHTOF(functions\fnc_cprLocal.sqf);
+            };
+			class treatment {
+                file = QPATHTOF(functions\fnc_treatment.sqf);
```
why this is in the pharma addon? this is made already in the circulation addon folder.

Tomcat-SG commented on addons/pharma/CfgFunctions.hpp:
```diff
@@ -0,0 +1,32 @@
+			class treatment {
+                file = QPATHTOF(functions\fnc_treatment.sqf);
+            };
+            class fullHealLocal {
```
why this is in the pharma addon? this is made already in the circulation addon folder.

Tomcat-SG commented on addons/pharma/CfgMagazines.hpp:
```diff
@@ -0,0 +1,36 @@
+
+        displayName = CSTRING(Carbonate_Box_Display);
+        descriptionShort = CSTRING(Carbonate_DescShort);
+        model = "\A3\Structures_F_EPA\Items\Medical\Painkillers_F.p3d";
```
Do we need some new 3d models? I have a modder in my server team, who would do some models for us if needed. Same for all the other medications ^^

Tomcat-SG commented on addons/pharma/CfgMagazines.hpp:
```diff
@@ -0,0 +1,36 @@
+class CfgMagazines {
+    class CA_Magazine;
+
+    class kat_Painkiller: CA_Magazine {
```
If youre adding the kat_painkiller to the pharma adddon, then you should remove them from the circulation mod. Im not sure, but i dont see deleted kat_painkiller in this pr for the circulation mod. Correct me if im wrong

Tomcat-SG commented on addons/pharma/CfgVehicles.hpp:
```diff
@@ -0,0 +1,40 @@
+        ace_cargo_canLoad = 1;
+    };
+
+    class kat_PainkillerItem: WeaponHolder_Single_limited_item_F {
```
If youre adding the kat_painkiller to the pharma adddon, then you should remove them from the circulation mod. Im not sure, but i dont see deleted kat_painkiller in this pr for the circulation mod. Correct me if im wrong

Tomcat-SG commented on addons/pharma/CfgWeapons.hpp:
```diff
@@ -0,0 +1,131 @@
+    class ACE_ItemCore;
+    class CBA_MiscItem_ItemInfo;
+
+    class kat_PainkillerItem: ACE_ItemCore {
```
If youre adding the kat_painkiller to the pharma adddon, then you should remove them from the circulation mod. Im not sure, but i dont see deleted kat_painkiller in this pr for the circulation mod. Correct me if im wrong

Tomcat-SG commented on addons/pharma/XEH_postInit.sqf:
```diff
@@ -0,0 +1,11 @@
+#include "script_component.hpp"
+
+["treatmentIVfalse", {_this call FUNC(treatmentAdvanced_IV)}] call CBA_fnc_addEventHandler;
```
Function isnt existing

Tomcat-SG commented on addons/pharma/XEH_postInit.sqf:
```diff
@@ -0,0 +1,11 @@
+#include "script_component.hpp"
+
+["treatmentIVfalse", {_this call FUNC(treatmentAdvanced_IV)}] call CBA_fnc_addEventHandler;
+["IVreset", {_this call FUNC(removeEffect_IV)}] call CBA_fnc_addEventHandler;
```
Function isnt existing

Tomcat-SG commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -0,0 +1,104 @@
+    QGVAR(enable),
+    "CHECKBOX",
+    LLSTRING(SETTING_ENABLE),
+    "KAT - ADV Medical: Pharmacy",
```
We should add a specific Description like in all other KAT Addons

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,73 @@
+
+_patient setVariable [QGVAR(IVplaced), true, true];
+
+if (_usedItem == "kat_IV_16") then {
```
```suggestion
if (_usedItem isEqualTo "kat_IV_16") then {
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        items[] = {"kat_IO_FAST"};
+        condition = QUOTE(!(_patient getVariable [ARR_2(QQGVAR(IVplaced), false)]));
+        treatmentTime = QGVAR(IOestablish);
+        callbackSuccess = QFUNC(applyIV);
```
Missing params? Function wants these params `["_medic", "_patient", "_bodyPart", "", "", "_usedItem"]`

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        items[] = {"kat_IV_16"};
+        condition = QUOTE(!(_patient getVariable [ARR_2(QQGVAR(IVplaced), false)])) && QFUNC(checkIV);
+        treatmentTime = QGVAR(IVestablish);
+        callbackSuccess = QFUNC(applyIV);
```
Missing params? Function wants these params `["_medic", "_patient", "_bodyPart", "", "", "_usedItem"]`

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,73 @@
+};
+
+[{
+    private _patient = _this select 0;
```
from which called parameter?

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,73 @@
+		_patient setVariable [QGVAR(flowRate), 0, true]
+	} else {
+		_patient setVariable [QGVAR(flowRate), ace_medical_ivFlowRate, true];
+	};
```
Also please add this behavior under this line `_patient setVariable ["kat_IVPharma_PFH", nil];`

mazinskihenry replied:
Check to make sure I did this right please.

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,73 @@
+    private _patient = _this select 0;
+    [_patient, _patient] call FUNC(retrieveIV);
+}, [_patient], GVAR(IVdrop)] call CBA_fnc_waitAndExecute;
+
```
And also add this behavior pls `if (_unit getVariable ["kat_IVPharma_PFH", false]) exitWith {};
_unit setVariable ["kat_IVPharma_PFH", true];`

mazinskihenry replied:
Check to make sure I did this right please.

Tomcat-SG replied:
Nope. This isnt correct

Tomcat-SG commented on addons/pharma/functions/fnc_checkIV.sqf:
```diff
@@ -0,0 +1,34 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "LeftLeg"] call kat_circulation_fnc_applyIV;
```
not correct function name

Tomcat-SG commented on addons/pharma/functions/fnc_checkIV.sqf:
```diff
@@ -0,0 +1,34 @@
+ * 2: Body Part <STRING>
+ *
+ * Return Value:
+ * None
```
Return Value missing

Tomcat-SG commented on addons/pharma/functions/fnc_checkIV.sqf:
```diff
@@ -0,0 +1,34 @@
+#include "script_component.hpp"
+/*
+ * Author: 2LT.Mazinski
+ * Opens an IV/IO on a patient and changes the patient's flow variable
```
Same text as applyIV?

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        category = "advanced";
+        allowedSelections[] = {"LeftArm", "RightArm", "LeftLeg", "RightLeg"};
+        items[] = {"kat_IV_16"};
+        condition = QUOTE(!(_patient getVariable [ARR_2(QQGVAR(IVplaced), false)])) && QFUNC(checkIV);
```
Missing params? Function wants these params `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG replied:
```suggestion
        condition = QUOTE((!(_patient getVariable [ARR_2(QQGVAR(IVplaced), false)])) && FUNC(checkIV));
```

Tomcat-SG commented on addons/pharma/functions/fnc_getBloodVolumeChange.sqf:
```diff
@@ -0,0 +1,60 @@
+    _bloodBags = _bloodBags apply {
+        _x params ["_bagVolumeRemaining", "_type", "_bodyPart"];
+
+        if (_tourniquets select _bodyPart == 0) then {
```
```suggestion
        if (_tourniquets select _bodyPart isEqualTo 0) then {
```

Tomcat-SG commented on addons/pharma/functions/fnc_init.sqf:
```diff
@@ -0,0 +1,26 @@
+ * None
+ *
+ * Example:
+ * [player] call kat_circulation_fnc_init;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_removeIV.sqf:
```diff
@@ -0,0 +1,31 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "LeftLeg"] call ace_medical_treatment_fnc_tourniquetRemove
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_removeIV.sqf:
```diff
@@ -0,0 +1,31 @@
+private _IV = _patient getVariable [QGVAR(IVsite), 0];
+private _return = false;
+
+if ((_IV == _partIndex) && (_patient getVariable [QGVAR(IVplaced), false])) then {
```
```suggestion
if ((_IV isEqualTo _partIndex) && (_patient getVariable [QGVAR(IVplaced), false])) then {
```

Tomcat-SG commented on addons/pharma/functions/fnc_removeIV.sqf:
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: Glowbal, mharis001
+ * Removes the tourniquet from the patient on the given body part.
```
Wrong description

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+    class BloodIV: BasicBandage {
+        allowedSelections[] = {"Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg"};
+        medicRequired = QUOTE(ace_medical_medicIV);
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
```
Missing params? Function want these parameter `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG replied:
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && FUNC(removeIV));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = QGVAR(PushTime);
+        items[] = {"kat_TXA"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QGVAR(txaActive) && QFUNC(removeIV);
```
Missing params? Function want these parameter `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG replied:
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && GVAR(txaActive) && FUNC(removeIV));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = QGVAR(PushTime);
+        items[] = {"kat_norepinephrine"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
```
Missing params? Function want these parameter `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG replied:
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && FUNC(removeIV));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = QGVAR(PushTime);
+        items[] = {"kat_nitroglycerin"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
```
Missing params? Function want these parameter `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG replied:
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && FUNC(removeIV));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        allowedSelections[] = {"Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg"};
+        treatmentTime = 3;
+        items[] = {};
+        condition = QFUNC(removeIV);
```
Missing params? Function want these parameter `["_medic", "_patient", "_bodyPart"]`

Tomcat-SG commented on addons/pharma/functions/fnc_removeItemfromMag.sqf:
```diff
@@ -0,0 +1,40 @@
+ *
+ * Public: No
+ */
+
```
Why do we need this file in the pharma addon, when it is already in the circulation mod? if you need it here, then call it like in the example

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+
+private _site = _patient getVariable [QGVAR(IVsite), 0];
+
+if (GVAR(IVreuse) == true) then {
```
```suggestion
if (GVAR(IVreuse)) then {
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+private _site = _patient getVariable [QGVAR(IVsite), 0];
+
+if (GVAR(IVreuse) == true) then {
+    if (_site == 1) then {
```
```suggestion
    if (_site isEqualTo 1) then {
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+private _blood = 0;
+private _plasma = 0;
+
+{
```
check the line spacing in this file for the switches

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        treatmentTime = 3;
+        items[] = {};
+        condition = QFUNC(removeIV);
+        callbackSuccess = QFUNC(retrieveIV);
```
Missing params? Function want these parameter `["_medic", "_patient"]`

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+#include "script_component.hpp"
+/*
+ * Author: 2LT.Mazinski
+ * Opens an IV/IO on a patient and changes the patient's flow variable
```
Same text as applyIV?

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Atropine.sqf:
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Atropine.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_AtropineLocal.sqf:
```diff
@@ -0,0 +1,34 @@
+#include "script_component.hpp"
+/*
+ * Author: 2LT.Mazinski
+ * Locates and Removes 1x Morphine after the administration of Naloxone.
```
This is the Atropine file, so whiy it removes 1x Morphine after Naloxone?

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_AtropineLocal.sqf:
```diff
@@ -0,0 +1,34 @@
+ * None
+ *
+ * Example:
+ * [player, "Atropine"] call kat_circulation_fnc_treatmentAdvanced_NaloxoneLocal;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_AtropineLocal.sqf:
```diff
@@ -0,0 +1,34 @@
+ * 1: Medication <STRING>
+ *
+ * Return Value:
+ * None
```
Return Value Missing

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_AtropineLocal.sqf:
```diff
@@ -0,0 +1,34 @@
+{
+    _x params ["_medication"];
+
+    if (_medication == "BRADYCARDIA") exitWith {
```
```suggestion
    if (_medication isEqualTo "BRADYCARDIA") exitWith {
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Carbonate.sqf:
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Carbonate.sqf:
```diff
@@ -0,0 +1,29 @@
+ *
+ * Arguments:
+ * 0: Medic <OBJECT>
+ * 1: Patient <OBJECT>
```
3rd parameter is missing

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Carbonate.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Carbonate.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_CarbonateLocal.sqf:
```diff
@@ -0,0 +1,33 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding local
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_CarbonateLocal.sqf:
```diff
@@ -0,0 +1,33 @@
+ * Succesful treatment <BOOL>
+ *`
+ * Example:
+ * [player, "Accuvac"] call kat_airway_fnc_treatmentAdvanced_accuvacLocal;
```
wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Generic.sqf:
```diff
@@ -0,0 +1,28 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Generic.sqf:
```diff
@@ -0,0 +1,28 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Generic.sqf:
```diff
@@ -0,0 +1,28 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_GenericLocal.sqf:
```diff
@@ -0,0 +1,25 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_GenericLocal.sqf:
```diff
@@ -0,0 +1,25 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_GenericLocal.sqf:
```diff
@@ -0,0 +1,25 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Naloxone.sqf:
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Naloxone.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Naloxone.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_NaloxoneLocal.sqf:
```diff
@@ -0,0 +1,36 @@
+ * 1: Medication <STRING>
+ *
+ * Return Value:
+ * None
```
Return Value missing

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_NaloxoneLocal.sqf:
```diff
@@ -0,0 +1,36 @@
+{
+    _x params ["_medication"];
+
+    if (_medication == "Morphine") exitWith {
```
```suggestion
    if (_medication isEqualTo "Morphine") exitWith {
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Ondansetron.sqf:
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Ondansetron.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
Wrong function Name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Ondansetron.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_OndansetronLocal.sqf:
```diff
@@ -0,0 +1,32 @@
+#include "script_component.hpp"
+/*
+ * Author: 2LT.Mazinski
+ * Locates and Removes 1x Morphine after the administration of Naloxone.
```
Wrong description?

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_OndansetronLocal.sqf:
```diff
@@ -0,0 +1,32 @@
+ * 0: Medic <OBJECT>
+ * 1: Medication <STRING>
+ *
+ * Return Value:
```
Missing returnVlaue

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_OndansetronLocal.sqf:
```diff
@@ -0,0 +1,32 @@
+ * None
+ *
+ * Example:
+ * [player, "Ondansetron"] call kat_circulation_fnc_treatmentAdvanced_NaloxoneLocal;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_OndansetronLocal.sqf:
```diff
@@ -0,0 +1,32 @@
+_target setVariable [QGVAR(ondUse), true, true];
+
+[{
+    private _patient = _this select 0;
```
why, you have the param _target already

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_OndansetronLocal.sqf:
```diff
@@ -0,0 +1,32 @@
+
+[_target, _item] call ace_medical_treatment_fnc_addToTriageCard;
+[_target, "activity", LSTRING(push_log), [[_medic] call ace_common_fnc_getName, "Ondansetron"]] call ace_medical_treatment_fnc_addToLog;
+[_target, "TXA", 5, 120, 0, 0, 0] call ace_medical_status_fnc_addMedicationAdjustment;
```
Only a question, because im not a medical pro. Is Ondansetron Equal to TXA? Because it is called in the file both...

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Reorientation.sqf:
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong description

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Reorientation.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
Wrong functionname

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_Reorientation.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_ReorientationLocal.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Arguments:
+ * 0: Medic <OBJECT>
+ * 1: Patient <OBJECT>
+ * 2: Item classname <STRING>
```
you only have 2 params in your code

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_ReorientationLocal.sqf:
```diff
@@ -0,0 +1,29 @@
+ * Succesful treatment <BOOL>
+ *`
+ * Example:
+ * [player, "Carbonate"] call kat_airway_fnc_treatmentAdvanced_CarbonateLocal;
```
wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXA.sqf:
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Katalam
+ * Airway Management for occluding
```
Wrong desscription?

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXA.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Succesful treatment started <BOOL>
+ *
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
```
Wrong function name

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXA.sqf:
```diff
@@ -0,0 +1,27 @@
+ * Example:
+ * [player, cursorTarget] call kat_airway_fnc_treatmentAdvanced_accuvac;
+ *
+ * Public: Yes
```
```suggestion
 * Public: No
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,65 @@
+ * 1: Medication <STRING>
+ *
+ * Return Value:
+ * None
```
Missing return Value

Tomcat-SG replied:
Missing returnValue

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,65 @@
+[_target, "activity", LSTRING(push_log), [[_medic] call ace_common_fnc_getName, "TXA"]] call ace_medical_treatment_fnc_addToLog;
+[_target, "TXA", 5, 120, 0, 0, 0] call ace_medical_status_fnc_addMedicationAdjustment;
+
+
```
Also add this behavior for the pfh `if (_unit getVariable ["kat_TXA_PFH", false]) exitWith {};
_unit setVariable ["kat_TXA_PFH", true];`

mazinskihenry replied:
Check that I did this right please.

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,65 @@
+    private _alive = alive _target;
+
+    if ((!_alive) || (_TXA == false)) exitWith {
+        [_idPFH] call CBA_fnc_removePerFrameHandler; 
```
ALso add here this behavior for the pfh `_unit setVariable ["kat_TXA_PFH", nil];`

mazinskihenry replied:
Check that I did this right please.

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,65 @@
+
+    private _alive = alive _target;
+
+    if ((!_alive) || (_TXA == false)) exitWith {
```
```suggestion
    if ((!_alive) || (!_TXA)) exitWith {
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,65 @@
+    {
+        _x params ["_id", "_bodyPart", "_amount"];
+
+        if (_once == false && _id != 20 && _amount > 0) exitWith {
```
```suggestion
        if (!_once && (_id != 20) && (_amount > 0)) exitWith {
```

Tomcat-SG commented on addons/pharma/ui/CrossPanel_Dialog.hpp:
```diff
@@ -0,0 +1,25 @@
+    idd = -1;
+    movingEnable = 1;
+    onLoad = QUOTE(uiNamespace setVariable [ARR_2(QQGVAR(CrossPanel_Display), (_this select 0))]);
+    onUnload = QUOTE(_this call FUNC(onCloseDialog));
```
Where is this function defined?

Tomcat-SG reviewed (changes requested):
- [x] Suggestions
- [x] handleBreathing are a lot of things, which should be changed. I will add a corrected file to discord.
- [x] Maybe there are more errors. So please check all the files again, maybe you find something else, which i didnt find yet

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        displayNameProgress = CSTRING(X_Remove_Action_Progress);
         items[] = {};
-        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(X), true)]);
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(X), true)]) && QUOTE(!(_patient getVariable [ARR_2(QQGVAR(vehicleTrue), false)]));
```
```suggestion
        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(X), true)] && !(_patient getVariable [ARR_2(QQGVAR(vehicleTrue), false)]));
```

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        displayName = "Vehicle: Attach AED-X";
+        displayNameProgress = CSTRING(X_Action_Progress);
+        items[] = {};
+        condition = QUOTE(!(_patient getVariable [ARR_2(QQGVAR(X), false)])) && QFUNC(vehicleCheck);
```
```suggestion
        condition = QUOTE(!(_patient getVariable [ARR_2(QQGVAR(X), false)]) && FUNC(vehicleCheck));
```

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        items[] = {};
+        callbackSuccess = "[_medic, _patient, 'AED-X'] call kat_circulation_fnc_AEDSuccess; _patient setVariable ['kat_AEDinUse', false, true];";
+        //condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(X), false)]) || [ARR_2(_medic, 'kat_AED')] call ace_common_fnc_hasItem || [ARR_2(_medic, 'kat_X_AED')] call ace_common_fnc_hasItem);
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(vehicleTrue), true)]) && QUOTE(_patient getVariable [ARR_2(QQGVAR(X), true)]);
```
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(vehicleTrue), true)]) && (_patient getVariable [ARR_2(QQGVAR(X), true)]));
```

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -122,18 +117,53 @@ class ACE_Medical_Treatment_Actions {
+        displayName = "Vehicle: Remove AED-X";
+        displayNameProgress = CSTRING(X_Remove_Action_Progress);
+        items[] = {};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(vehicleTrue), true)]) && QUOTE(_patient getVariable [ARR_2(QQGVAR(X), true)]);
```
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(vehicleTrue), true)]) && (_patient getVariable [ARR_2(QQGVAR(X), true)]));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = QGVAR(PushTime);
+        items[] = {"kat_phenylephrine"};
+        condition = QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && QFUNC(removeIV);
```
Missing Params for removeIV?

Tomcat-SG replied:
```suggestion
        condition = QUOTE((_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]) && FUNC(removeIV));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = 5;
+        items[] = {"kat_amiodarone"};
+        condition = QUOTE(!([_patient] call ace_common_fnc_isAwake)) && QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]);
```
```suggestion
        condition = QUOTE((!([_patient] call ace_common_fnc_isAwake)) && (_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]));
```

Tomcat-SG commented on addons/pharma/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -0,0 +1,189 @@
+        medicRequired = 1;
+        treatmentTime = 5;
+        items[] = {"kat_lidocaine"};
+        condition = QUOTE(!([_patient] call ace_common_fnc_isAwake)) && QUOTE(_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]);
```
```suggestion
        condition = QUOTE((!([_patient] call ace_common_fnc_isAwake)) && (_patient getVariable [ARR_2(QQGVAR(IVplaced), true)]));
```

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/ACE_Medical_Treatment_Actions.hpp

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/airway/functions/fnc_updateInjuryList.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/airway/functions/fnc_updateInjuryList.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/airway/functions/fnc_updateInjuryList.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/airway/functions/fnc_updateInjuryList.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/airway/functions/fnc_updateInjuryList.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/functions/fnc_AEDanalyze.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/functions/fnc_AEDanalyze.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Update addons/circulation/functions/fnc_cprLocal.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

mazinskihenry committed:
Post-Review Update

mazinskihenry committed:
Update fnc_handleBreathing.sqf

mazinskihenry committed:
Disable Rhythm System

DIsables the advanced rhythms

mazinskihenry committed:
Fixes to SpO2 and AED-X

mazinskihenry committed:
Fixes

mazinskihenry committed:
Last time

mazinskihenry committed:
Update XEH_preInit.sqf

mazinskihenry committed:
AGAIN

mazinskihenry committed:
Update fnc_handleBreathing.sqf

mazinskihenry committed:
Final updates

mazinskihenry committed:
Update stringtable.xml

mazinskihenry committed:
Dev Update

mazinskihenry committed:
Update fnc_handleBreathing.sqf

mazinskihenry committed:
Update fnc_handleBreathing.sqf

mazinskihenry committed:
Update fnc_handleBreathing.sqf

mazinskihenry committed:
Updated fixes

mazinskihenry committed:
Update XEH_preInit.sqf

mazinskihenry committed:
Update XEH_preInit.sqf

YetheSamartaka committed:
hyperextending head fix + hr conditions + perf improvement

added missing hyperextending head if for the SpO2
added new conditions for heart rate so it is not overlapping with other possible states
improved performance by removing unnecessary "== true"

mazinskihenry committed:
I love mod development

mazinskihenry committed:
Update fnc_handleBreathing.sqf

YetheSamartaka committed:
_overstretch update

-returned _overstretch missing condition
-increased SpO2 raise value from 0.1 to 0.15 so it is closer to the original idea

YetheSamartaka committed:
overstreach timer increasement

There is a 120 seconds default timer that causes hyperextension to cancel out.
- changed 120 seconds timer to 600 seconds

YetheSamartaka committed:
pneumothorax medical menu fix

-fixes pneumothorax duplicity in the medical menu

YetheSamartaka committed:
SpO2 missing check put back

-puts back missing SpO2 check for the values below 75 when units are conscious. Currently, this is crucial for the XYZthoraxes to work. Possible other future features will require it as well.

Tomcat-SG commented on addons/circulation/ACE_Medical_Treatment_Actions.hpp:
```diff
@@ -32,12 +16,25 @@ class ACE_Medical_Treatment_Actions {
     };
+    class CheckRhythm: CheckPulse { 
+        displayName = CSTRING(Rhythm);
+        displayNameProgress = "Checking Rhythm";
```
Stringtable!!!

Tomcat-SG commented on addons/circulation/functions/fnc_AEDanalyze.sqf:
```diff
@@ -0,0 +1,55 @@
+	};
+};
+
+if ((GVAR(AdvRhythm) == false)) then {
```
```suggestion
if !(GVAR(AdvRhythm)) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -61,16 +103,40 @@ switch (_reviveObject) do {
-	if (_random <= _chance) then {
+	_chance = _chance + (_amiBoost + _lidoBoost * _epiBoost);
+
+    if ((_random <= _chance) && (_asystole == 1)) then {
```
```suggestion
    if ((_random <= _chance) && (_asystole isEqualTo 1)) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_cprLocal.sqf:
```diff
@@ -61,16 +103,40 @@ switch (_reviveObject) do {
+
 	if (_random <= _chance) then {
-		["ace_medical_CPRSucceeded", _patient] call CBA_fnc_localEvent;
+		if ((_randomAmi > 2) && (_asystole == 2)) then {
```
```suggestion
		if ((_randomAmi > 2) && (_asystole isEqualTo 2)) then {
```

Tomcat-SG commented on addons/circulation/functions/fnc_returnAED_X.sqf:
```diff
@@ -17,14 +17,23 @@
+    _patient setVariable [QGVAR(asystole), 0, true];
+};
+
+if (_AEDreturn == true) then {
```
```suggestion
if (_AEDreturn) then {
```

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,71 @@
+    	};
+    	case "rightleg": {_patient setVariable [QGVAR(IVsite), 5, true];
+    	};
+    };
```
```suggestion
    switch (_bodyPart) do {
    	case "leftarm": {
    	    _patient setVariable [QGVAR(IVsite), 2, true];
    	};
    	case "rightarm": {
    	    _patient setVariable [QGVAR(IVsite), 3, true];
    	};
    	case "leftleg": {
    	    _patient setVariable [QGVAR(IVsite), 4, true];
    	};
    	case "rightleg": {
    	    _patient setVariable [QGVAR(IVsite), 5, true];
    	};
    };
```

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,71 @@
+};
+
+if (_patient getVariable ["kat_IVPharma_PFH", false]) exitWith {};
+_patient setVariable ["kat_IVPharma_PFH", true];
```
```suggestion
```

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,71 @@
+ */
+
+params ["_medic", "_patient", "_bodyPart", "_usedItem"];
+
```
```suggestion

if (_patient getVariable ["kat_IVPharma_PFH", false]) exitWith {};
_patient setVariable ["kat_IVPharma_PFH", true];

```

Tomcat-SG commented on addons/pharma/functions/fnc_applyIV.sqf:
```diff
@@ -0,0 +1,71 @@
+        private _bloodBags = _patient getVariable ["ace_medical_ivBags", []];
+
+        if (_bloodBags isEqualTo []) then {
+            [_idPFH] call CBA_fnc_removePerFrameHandler;
```
```suggestion
            [_idPFH] call CBA_fnc_removePerFrameHandler;
            _unit setVariable ["kat_IVPharma_PFH", nil];
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+        case ("Plasma"): {
+        _plasma = _plasma + _volumeRemaining;
+        };
+    };
```
```suggestion
    switch (_type) do {
        case ("Saline"): {
            _saline = _saline + _volumeRemaining;
        };
        case ("Blood"): {
            _blood = _blood + _volumeRemaining;
        };
        case ("Plasma"): {
            _plasma = _plasma + _volumeRemaining;
        };
    };
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+        	};
+        	case (_saline > 150): {
+        	_medic addItem "ACE_salineIV_250";
+        	};
```
```suggestion
        	case (_saline > 1200): {
        	    _medic addItem "ACE_salineIV";
        	    _medic addItem "ACE_salineIV_500";
        	};
        	case (_saline > 800): {
        	    _medic addItem "ACE_salineIV";
        	};
        	case (_saline > 600): {
        	    _medic addItem "ACE_salineIV_500";
        	    _medic addItem "ACE_salineIV_250";
        	};
        	case (_saline > 400): {
        	    _medic addItem "ACE_salineIV_500";
        	};
        	case (_saline > 150): {
        	    _medic addItem "ACE_salineIV_250";
        	};
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+        	case (_blood > 150): {
+        	_medic addItem "ACE_bloodIV_250";
+        	};
+        };
```
```suggestion
        	case (_blood > 1200): {
        	    _medic addItem "ACE_bloodIV";
        	    _medic addItem "ACE_bloodIV_500";
        	};
        	case (_blood > 800): {
        	    _medic addItem "ACE_bloodIV";
        	};
        	case (_blood > 600): {
        	    _medic addItem "ACE_bloodIV_500";
        	    _medic addItem "ACE_bloodIV_250";
        	};
        	case (_blood > 400): {
        	    _medic addItem "ACE_bloodIV_500";
        	};
        	case (_blood > 150): {
        	    _medic addItem "ACE_bloodIV_250";
        	};
        };
```

Tomcat-SG commented on addons/pharma/functions/fnc_retrieveIV.sqf:
```diff
@@ -0,0 +1,123 @@
+        	};
+        	case (_plasma > 150): {
+        	_medic addItem "ACE_plasmaIV_250";
+        	};
```
```suggestion
        	case (_plasma > 1200): {
        	    _medic addItem "ACE_plasmaIV";
        	    _medic addItem "ACE_plasmaIV_500";
        	};
        	case (_plasma > 800): {
        	    _medic addItem "ACE_plasmaIV";
        	};
        	case (_plasma > 600): {
        	    _medic addItem "ACE_plasmaIV_500";
        	    _medic addItem "ACE_plasmaIV_250";
        	};
        	case (_plasma > 400): {
        	    _medic addItem "ACE_plasmaIV_500";
        	};
        	case (_plasma > 150): {
        	    _medic addItem "ACE_plasmaIV_250";
        	};
```

Tomcat-SG commented on addons/pharma/functions/fnc_treatmentAdvanced_TXALocal.sqf:
```diff
@@ -0,0 +1,68 @@
+ * 1: Medication <STRING>
+ *
+ * Return Value:
+ * None
```
```suggestion
 * Succesful treatment started <BOOL>
```

Tomcat-SG reviewed (changes requested):
If this and all the languages for the stringtables are ready, then we can merge it and upload it to the stable version of steam

YetheSamartaka committed:
TXALocal description

Successful treatment started <BOOL>

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
_AEDreturn perf improvement

removed "== true" for faster exec time

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
fnc_cprLocal utilise isEqualTo

Utilise isEqualTo for faster exec time

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
fnc_cprLocal.sqf isEqualTo

Utilize isEqualTo for faster exec time

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
advRhythm exec time and code clarity impr

removed negation and "== false" to improve exec time and code clarity

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
formating of fnc_retrieveIV.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
formating of fnc_retrieveIV.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
formating of fnc_retrieveIV.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
formating of fnc_retrieveIV.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
formating of fnc_applyIV.sqf

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>

YetheSamartaka committed:
applyIV PFH fix

Co-authored-by: Tomcat <51529018+Tomcat-SG@users.noreply.github.com>
